### DISCUSSION TITLE
Visualizing when one Job invokes another Job.

### DISCUSSION THREAD

**Discussion by lbstoller** [dagster-io/dagster#12025] [Q&A]
I would like to compose Jobs, by calling execute_job() from one job to a job in another file. This works fine, but the web UI does not indicate what has happened. I have to go find the external job run and look to see what happened (and to see the visualization). Is there a way (or a plan) to combine this type of workflow in the UI into a single place that can be inspected?

Thanks

**Comment by jamiedemaria** [comment]
Hi @lbstoller! In general, composing jobs like this is not considered best practice. We recommend composing graphs within a job, and then you will see the dependencies

here's an example

```


@op(out=Out(Int))
def emit_one():
    return 1


@op(ins={"numbers": In(List[Int])}, out=Out(Int))
def add(numbers):
    return sum(numbers)


@op(ins={"num": In(Float)}, out=Out(Float))
def div_two(num):
    return num / 2


@graph
def emit_two():
    return add([emit_one(), emit_one()])


@graph
def add_four_and_three():
    return add([emit_two(), emit_two(), 3])


@graph
def div_four(num):
    return div_two(num=div_two(num))


@op(ins={"num": In(Int)}, out=Out(Float))
def int_to_float(num):
    return float(num)


@job
def composition():
    div_four(int_to_float(add_four_and_three()))


```

If you want to run any of the subgraphs individually, you can make jobs for those individual graphs as well

**Comment by lbstoller** [comment]
Thanks. I am decomposing things into files that have a job for the work done in that file, and a graph that that I can reference from a master file (job). This seems to be working out okay.
